File: P2/PerfectNumber.cpp
#include <iostream>

using namespace std;
int F[100000];



int main(){
        
        int n, count = 0, i = 18, aux,sum;
    
        cin >> n;
        
        
        while (count < n)
        {
            i++;
            aux = i;
            sum = 0;
            while(aux > 0){
                sum +=aux%10;
                aux /= 10;
                
            }
            if(sum == 10)
                count++;
        }
        cout << i << endl;
}

File: P2/PerfectNumber.py
n = int(input())


soma_digitos = lambda a : sum([int(i) for i in list(str(a))])

n = 1
while n < 100:
    i = 0
    count = 0
    while count < n:
        i+=1
        if soma_digitos(i) == 10:
            count += 1



    #PA com a0 = 19 e razão 9
    an = 19 + (n-1)*9
    #print(an)

    print(n,i,an, i-an)
    n+=1